src/java/org/apache/cassandra/utils/streamhist/StreamingTombstoneHistogramBuilder.java

```diff
@@ -100,7 +100,7 @@ public void update(int point, int value)
     {
         point = ceilKey(point, roundSeconds);
 
-        if (spool.capacity > 0)
+        if (spool != null && spool.capacity > 0)
```

Comment:
can spool go to null and cause a NPE? (i'm not sure what the concurrency possible here is…)

but i suspect this and the other `spool != null` conditional checks above would be safer if they did…
```
    Spool spool = this.spool; // don't lose the reference incase releaseBuffers() is called concurrently
    if (spool != null && spool.capacity > 0)
    {
        …
    }
```

Reply:
good point. In practice that never happens (I instrumented the code and studied it before doing this), but I agree it’s bad hygiene.